<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from irb6640_185_280.xacro          | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="abb_irb6640_185_280" xmlns:xacro="http://ros.org/wiki/xacro">
  <!-- link list -->
  <link name="base_link">
     <inertial>
      <mass value="300"/>
      <origin rpy="0 0 0" xyz="-0 0 .1"/>
      <inertia ixx="0.1315"  ixy="0"  ixz="0" iyy="0.1415" iyz="0" izz="0.0200" />
    </inertial>
    <collision name="collision">
      <geometry>
        <mesh filename="package://abb_irb6640_support/meshes/irb6640_185_280/collision/base_link.stl"/>
      </geometry>
    </collision>
    <visual name="visual">
      <geometry>
        <mesh filename="package://abb_irb6640_support/meshes/irb6640_185_280/visual/base_link.dae"/>
      </geometry>
      <material name="yellow">
        <color rgba="1 1 0 1"/>
      </material>
    </visual>
  </link>
  <link name="link_1">
    <inertial>
      <origin xyz="0.18 0.15 0.410" rpy="0 0 0"/>
      <mass value="411"/>
      <inertia ixx="0.1315"  ixy="0"  ixz="0" iyy="0.1415" iyz="0" izz="0.0200" />
    </inertial>
    <collision name="collision">
      <geometry>
        <mesh filename="package://abb_irb6640_support/meshes/irb6640_185_280/collision/link_1.stl"/>
      </geometry>
    </collision>
    <visual name="visual">
      <geometry>
        <mesh filename="package://abb_irb6640_support/meshes/irb6640_185_280/visual/link_1.dae"/>
      </geometry>
      <material name="yellow"/>
    </visual>
  </link>
  <link name="link_2">
    <inertial>
      <origin xyz="0.0 -0.1 0.55" rpy="1.57079632679 -1.57079632679 0"/>
      <mass value="121"/>
      <inertia ixx="0.02"  ixy="0"  ixz="0" iyy="0.3962" iyz="0" izz="0.3952" />
    </inertial>
    <collision name="collision">
      <geometry>
        <mesh filename="package://abb_irb6640_support/meshes/irb6640_185_280/collision/link_2.stl"/>
      </geometry>
    </collision>
    <visual name="visual">
      <geometry>
        <mesh filename="package://abb_irb6640_support/meshes/irb6640_185_280/visual/link_2.dae"/>
      </geometry>
      <material name="yellow"/>
    </visual>
  </link>
  <link name="link_3">
    <inertial>
      <origin xyz="0.18 0.11 0.01" rpy="0 0 0"/>
      <mass value="113.4"/>
      <inertia ixx="0.045"  ixy="0"  ixz="0" iyy="0.04" iyz="0" izz="0.012" />
    </inertial>
    <collision name="collision">
      <geometry>
        <mesh filename="package://abb_irb6640_support/meshes/irb6640_185_280/collision/link_3.stl"/>
      </geometry>
    </collision>
    <visual name="visual">
      <geometry>
        <mesh filename="package://abb_irb6640_support/meshes/irb6640_185_280/visual/link_3.dae"/>
      </geometry>
      <material name="yellow"/>
    </visual>
  </link>
  <link name="link_4">
    <inertial>
      <origin xyz="0.41 0.0 0.0" rpy="0 1.57079632679 0"/>
      <mass value="74.6"/>
      <inertia ixx="0.437"  ixy="0"  ixz="0" iyy="0.447" iyz="0" izz="0.02" />
    </inertial>
    <collision name="collision">
      <geometry>
        <mesh filename="package://abb_irb6640_support/meshes/irb6640_185_280/collision/link_4.stl"/>
      </geometry>
    </collision>
    <visual name="visual">
      <geometry>
        <mesh filename="package://abb_irb6640_support/meshes/irb6640_185_280/visual/link_4.dae"/>
      </geometry>
      <material name="yellow"/>
    </visual>
  </link>
  <link name="link_5">
    <inertial>
      <origin xyz="0.0 0.05 0.0" rpy="0 0 0"/>
      <mass value="24.6"/>
      <inertia ixx="0.0133"  ixy="0"  ixz="0" iyy="0.025" iyz="0" izz="0.02" />
    </inertial>
    <collision name="collision">
      <geometry>
        <mesh filename="package://abb_irb6640_support/meshes/irb6640_185_280/collision/link_5.stl"/>
      </geometry>
    </collision>
    <visual name="visual">
      <geometry>
        <mesh filename="package://abb_irb6640_support/meshes/irb6640_185_280/visual/link_5.dae"/>
      </geometry>
      <material name="yellow"/>
    </visual>
  </link>
  <link name="link_6">
    <inertial>
      <origin xyz="0.0 0.0 0.02" rpy="0 0 0"/>
      <mass value="5"/>
      <inertia ixx="0.02"  ixy="0"  ixz="0" iyy="0.02" iyz="0" izz="0.0253" />
    </inertial>
    <collision name="collision">
      <geometry>
        <mesh filename="package://abb_irb6640_support/meshes/irb6640_185_280/collision/link_6.stl"/>
      </geometry>
    </collision>
    <visual name="visual">
      <geometry>
        <!-- <mesh filename="package://abb_irb6640_support/meshes/irb6640_185_280/visual/link_6.dae"/> -->
        <!-- <box size=".5 .5 .5" /> -->
        <sphere radius="0.5"/>
      </geometry>
      <material name="yellow"/>
    </visual>
  </link>
  <!--
  <link name="tool0"/>
-->
  <!--Cylinder and piston -->
  <!--
  <link name="link_cylinder">
    <collision name="collision">
      <geometry>
        <mesh filename="package://abb_irb6640_support/meshes/irb6640_185_280/collision/cylinder_link.stl"/>
      </geometry>
    </collision>
    <visual name="visual">
      <geometry>
        <mesh filename="package://abb_irb6640_support/meshes/irb6640_185_280/visual/cylinder_link.dae"/>
      </geometry>
      <material name="yellow"/>
    </visual>
  </link>
  <link name="link_piston">
    <collision name="collision">
      <geometry>
        <mesh filename="package://abb_irb6640_support/meshes/irb6640_185_280/collision/piston_link.stl"/>
      </geometry>
    </collision>
    <visual name="visual">
      <geometry>
        <mesh filename="package://abb_irb6640_support/meshes/irb6640_185_280/visual/piston_link.dae"/>
      </geometry>
      <material name="yellow"/>
    </visual>
  </link>
-->
  <!-- end of link list -->
  <!-- joint list -->
  <joint name="joint_1" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.780"/>
    <axis xyz="0 0 1"/>
    <parent link="base_link"/>
    <child link="link_1"/>
    <limit effort="100000" lower="-2.967" upper="2.967" velocity="1.7453"/>
  </joint>
  <joint name="joint_2" type="revolute">
    <origin rpy="0 0 0" xyz="0.320 0 0"/>
    <axis xyz="0 1 0"/>
    <parent link="link_1"/>
    <child link="link_2"/>
    <limit effort="0" lower="-1.134" upper="1.4855" velocity="1.5707"/>
  </joint>
  <joint name="joint_3" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 1.075"/>
    <axis xyz="0 1 0"/>
    <parent link="link_2"/>
    <child link="link_3"/>
    <limit effort="0" lower="-3.142" upper="1.222" velocity="1.5707"/>
  </joint>
  <joint name="joint_4" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.2"/>
    <axis xyz="1 0 0"/>
    <parent link="link_3"/>
    <child link="link_4"/>
    <limit effort="0" lower="-5.236" upper="5.236" velocity="2.9671"/>
  </joint>
  <joint name="joint_5" type="revolute">
    <origin rpy="0 0 0" xyz="1.392 0 0 "/>
    <axis xyz="0 1 0"/>
    <parent link="link_4"/>
    <child link="link_5"/>
    <limit effort="0" lower="-2.094" upper="2.094" velocity="2.4435"/>
  </joint>
  <joint name="joint_6" type="revolute">
    <origin rpy="0 0 0" xyz="0.2 0 0 "/>
    <axis xyz="1 0 0"/>
    <parent link="link_5"/>
    <child link="link_6"/>
    <limit effort="0" lower="-6.283" upper="6.283" velocity="3.3161"/>
  </joint>
<!--
  <joint name="joint_6-tool0" type="fixed">
    <parent link="link_6"/>
    <child link="tool0"/>
    <origin rpy="0 1.57079632679 0" xyz="0 0 0"/>
  </joint>
-->
  <!--
  <joint name="joint_cylinder" type="continuous">
    <origin rpy="0 0 0" xyz="-0.3647 0 -0.1455"/>
    <axis xyz="0 1 0"/>
    <parent link="link_1"/>
    <child link="link_cylinder"/>
    <mimic joint="joint_2" multiplier="-0.25"/>
  </joint>
  <joint name="joint_piston" type="continuous">
    <origin rpy="0 0 0" xyz="-0.22 0 -0.0672"/>
    <axis xyz="0 1 0"/>
    <parent link="link_2"/>
    <child link="link_piston"/>
    <mimic joint="joint_2" multiplier="-1.25"/>
  </joint>
 -->
  <!-- end of joint list -->
  <!-- ROS base_link to ABB World Coordinates transform -->
  <!--  <link name="base"/>
  <joint name="base_link-base" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="base_link"/>
    <child link="base"/>
  </joint>
-->
</robot>

